src/main/target/CFGF4OMNI/target.h

```diff
@@ -0,0 +1,174 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#define USE_TARGET_CONFIG
+
+#define TARGET_BOARD_IDENTIFIER "F4OM"
+#define USBD_PRODUCT_STRING "Betaflight F4 Omni"
+
+#define LED0_PIN                PB5
+#define BEEPER                  PB4
+#define BEEPER_INVERTED
+
+#define USE_DSHOT_DMAR
+
+// These inverter control pins collide with timer channels on CH5 and CH6 pads.
+// Users of these timers/pads must un-map the inverter assignment explicitly.
+#define INVERTER_PIN_UART6      NONE // PC8 // Omnibus F4 V3 and later
+#define INVERTER_PIN_UART3      NONE // PC9 // Omnibus F4 Pro Corners
+
+#define USE_ACC
+#define USE_ACC_SPI_MPU6000
+
+#define USE_GYRO
+#define USE_GYRO_SPI_MPU6000
+
+#define MPU6000_CS_PIN          PA4
+#define MPU6000_SPI_INSTANCE    SPI1
+
+// MPU6000 interrupts
+#define USE_EXTI
+#define MPU_INT_EXTI            PC4
+#define USE_MPU_DATA_READY_SIGNAL
+
+#define GYRO_MPU6000_ALIGN       CW270_DEG
+#define ACC_MPU6000_ALIGN        CW270_DEG
+
+// Support for iFlight OMNIBUS F4 V3
+// Has ICM20608 instead of MPU6000
+// OMNIBUSF4SD is linked with both MPU6000 and MPU6500 drivers
+#define USE_ACC_SPI_MPU6500
+#define USE_GYRO_SPI_MPU6500
+#define MPU6500_CS_PIN          MPU6000_CS_PIN
+#define MPU6500_SPI_INSTANCE    MPU6000_SPI_INSTANCE
+#define GYRO_MPU6500_ALIGN      GYRO_MPU6000_ALIGN
+#define ACC_MPU6500_ALIGN       ACC_MPU6000_ALIGN
+
+#define USE_MAG
+#define USE_MAG_HMC5883
+#define MAG_HMC5883_ALIGN       CW90_DEG
+
+#define USE_BARO
+#define USE_BARO_SPI_BMP280
+#define BMP280_SPI_INSTANCE     SPI3
+#define BMP280_CS_PIN           NONE
+#define USE_BARO_BMP085
+#define USE_BARO_BMP280
+#define USE_BARO_MS5611
+#define BARO_I2C_INSTANCE       (I2CDEV_2)
+
+#define DEFAULT_BARO_SPI_BMP280
+
+#define USE_OSD
+#define USE_MAX7456
+#define MAX7456_SPI_INSTANCE    SPI3
+#define MAX7456_SPI_CS_PIN      PA15
+#define MAX7456_SPI_CLK         (SPI_CLOCK_STANDARD) // 10MHz
+#define MAX7456_RESTORE_CLK     (SPI_CLOCK_FAST)
+
+#define USE_SDCARD
+#define SDCARD_DETECT_INVERTED
+#define SDCARD_DETECT_PIN               PB7
+#define SDCARD_SPI_INSTANCE             SPI2
+#define SDCARD_SPI_CS_PIN               PB12
+// SPI2 is on the APB1 bus whose clock runs at 84MHz. Divide to under 400kHz for init:
+#define SDCARD_SPI_INITIALIZATION_CLOCK_DIVIDER 256 // 328kHz
+// Divide to under 25MHz for normal operation:
+#define SDCARD_SPI_FULL_SPEED_CLOCK_DIVIDER 4 // 21MHz
+
+#define SDCARD_DMA_CHANNEL_TX                   DMA1_Stream4
+#define SDCARD_DMA_CHANNEL_TX_COMPLETE_FLAG     DMA_FLAG_TCIF4
+#define SDCARD_DMA_CLK                          RCC_AHB1Periph_DMA1
+#define SDCARD_DMA_CHANNEL                      DMA_Channel_0
+
+#define USE_VCP
+#define VBUS_SENSING_PIN PC5
+
+#define USE_UART1
+#define UART1_RX_PIN            NONE // PA10
+#define UART1_TX_PIN            NONE // PA9
+
+#define USE_UART3
```

Comment:
Do we even need these defines? If so, this will not be universally configurable. (Imagine a board wanting to expose UART5.)

Reply:
No, these `NONE` lines are to prevent defaults to be configured.
When we finally go fully configurable, we will drop those defaults.

Reply:
Oh, defaults for UART pins seem to be NONE for F4, so we can just drop these lines.

Reply:
That's exactly where my proposal comes in: Instead of hanging on to 1000s of lines of define processing at compile time for target configuration, and then having to remove them all at once when they become untenable because we want to move to generic targets, we should start to move to a runtime configuration schema for things where this is easy already (like SOFTSERIAL).

Reply:
No. All these will be gone. No burden.

Reply:
Well, for them to 'be gone', somebody will have to manually strip them out, and then trace down all of the forgotten `#ifdef`s buried deep inside the code - the earlier we start with this, the more time we will have for these things to bubble up.

Reply:
What I suggested was really the first step of 'going post-flash', by starting to reduce the amount of compile time logic that is currently used to translate `target.h` into settings, because that will not be a viable approach when we want to go post-flash.

Reply:
The statically compiled `timerHardware` array will eventually be gone. Thoughts?

Reply:
Only that removing it will be straightforward, since there won't be the potential for stray `#ifdefs` left in unexpected places.